packages/mindful/marko-web/middleware/advertising-post-by-id-or-import-entity.js

```diff
@@ -0,0 +1,27 @@
+const { asyncRoute } = require('@parameter1/base-cms-utils');
+const { getAsObject } = require('@parameter1/base-cms-object-path');
+const defaultFragment = require('../../graphql/fragments/advertising-post-by-id');
+
+module.exports = (app, {
+  route,
+  template,
+  tenant,
+  provider,
+  fragment = defaultFragment,
+}) => {
+  app.get(route, asyncRoute(async (req, res) => {
+    const { service } = req.mindful;
+    const { id: _id } = req.params;
+    const { namespace } = tenant;
+    const result = await service.getAdvertisingPostByIdOrImportEntity({
```

Comment:
Rename the service method to `getAdvertisingPostAsNativeStory` and move the conversion to that method... see below.

In other words this should become
```js
const story = service.getAdvertisingPostAsNativeStory({ _id, preview: Boolean(req.query.preview), tenant });
res.marko(template, { story });
```